### PR TITLE
updated local and task types

### LINKED ISSUE #24: Rename `Task` Type Alias
The name is somewhat misleading, since it may not truly be a "task".

Rename:
 - `Local` to `LocalBoxNotifier`
 - `Task` to `BoxNotifier`

### PR DESCRIPTION
**Pull request by taylorjeandev**

### CODE REVIEW

**Comment by AldaronLau** [comment]
You'll also need to update each occurance where `Task` and `LocalTask` are used in other files to get building working and the tests to pass.

**Comment by taylorjeandev** [comment]
I made some changes in the exec.rs file, as well as a couple more in the lib.rs file. 

I took a look at the results of tests you ran, and it looked like the errors were coming from the lib.rs file.

File: src/exec.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 // At your choosing (See accompanying files LICENSE_APACHE_2_0.txt,
 // LICENSE_MIT.txt and LICENSE_BOOST_1_0.txt).
 
-use alloc::{sync::Arc, task::Wake};
```
Comment (AldaronLau) [inline src/exec.rs]: This line should be reverted back to how it was.  You can see all the errors on your machine by running `cargo test`.

File: src/exec.rs [hunk 2]
```diff
@@ -147,16 +147,16 @@ impl<I: Spawn> Drop for Executor<I> {
                     let mut tasks = t.take();
                     let output = tasks.pop();
                     t.set(tasks);
-                    if let Some(task) = output {
```
Comment (AldaronLau) [inline src/exec.rs]: This variable name should be changed back to `task`.

File: src/exec.rs [hunk 3]
```diff
@@ -147,16 +147,16 @@ impl<I: Spawn> Drop for Executor<I> {
...
+                        return Ready(BoxNotifier);
                     }
                     Pending
                 })
             }
         }
 
-        fn spawn(cx: &mut Tasks, task: Local<'static, ()>) -> Poll<()> {
-            cx.0.push(task);
+        fn spawn(cx: &mut Tasks, BoxNotifier: LocalBoxNotifier<'static, ()>) -> Poll<()> {
```
Comment (AldaronLau) [inline src/exec.rs]: Same with this variable name.

**Comment by taylorjeandev** [comment]
thank you for your patience and working with me on this! its very much appreciated

**Review by AldaronLau — APPROVED** [review]

**Comment by AldaronLau** [comment]
> thank you for your patience and working with me on this! its very much appreciated

Sure thing - thanks for taking on this task!  Looks good to merge once all the tests finish.

**Pull request merged**